rclpy/src/rclpy/_rclpy.c

```diff
@@ -2320,7 +2320,8 @@ rclpy_get_node_names(PyObject * Py_UNUSED(self), PyObject * args)
   size_t idx;
   for (idx = 0; idx < node_names.size; ++idx) {
     PyList_SetItem(
-      pynode_names, idx, PyUnicode_FromString(node_names.data[idx]));
+      pynode_names, idx, PyUnicode_FromString(
+        node_names.data[idx] ? node_names.data[idx] : ""));
```

Comment:
Is there a difference between a null pointer and an empty string coming from `rcl_get_node_names()`? If so, maybe this should return Py_None instead.

Reply:
Absolutely. Updated the patch to use `Py_None`.